### PR TITLE
Hue ota

### PR DESCRIPTION
**Pull request by henriknil**
Philips Hue OTA support for selected devices

### CODE REVIEW

**Comment by henriknil** [comment]
This might need some work before merging.
I don't know if it's possible to get support for or devices or possible implementing a fallback to zigbee-OTA.

**Comment by Koenkk** [comment]
Would be really nice if this could be enabled for all devices, what is blocking this? 

Pinging @thejulianjes and @mkaay here since they have previously updated Hue otas to the zigbee-ota repo.

**Comment by TheJulianJES** [comment]
To my knowledge, the ``https://firmware.meethue.com/v1/checkupdate`` endpoint is only used in the Hue Bluetooth app at the moment. This app will no longer be updated/supported soon-ish, as the main Hue app is already able to control supporting Philips Hue lights via Bluetooth (as well as Zigbee if the user has a Hue Bridge v2).
Because of this, the endpoint mentioned above only gives out updates for Bluetooth capable lights.
(I did not yet check which endpoint is used by the main Hue app when only Bluetooth devices are connected. It might be the same one from above.)
All links for OTA images coming from this endpoint start with ``https://firmware.meethue.com/storage/``.

If the Hue Bridge is used with the Hue system (which is required for older non-Bluetooth lights), the Bridge checks for the updates. The Hue Bridge, however, does not use the endpoint mentioned above. Instead, it uses "a more complicated API" which also requires sending "confidential" bridge-specific data. This basically makes it impossible to implement here.
Those links for OTA images seem to all start with ``http://fds.dc1.philips.com/firmware``.

In my experience, the updates from the Hue Bluetooth app (endpoint listed at the top) also sometimes differ from the updates provided via the Hue Bridge (most of the times, the app ones are a bit newer). I suspect this is to ensure stability for the bigger Hue Bridge/Zigbee user base and because the Bluetooth part isn't/wasn't as stable as the Zigbee part.

For ZHA, deCONZ and other users (who need to download all OTA files), it would be nice to have an up-to-date list of **all** Hue OTA files (which zigbee-OTA currently contains). I'm not sure if the approach in this PR would mean that some Hue image links would be "deleted" from the zigbee-OTA repo or at least get out of sync with images used by Z2M (through the mobile endpoint).
Using the mobile endpoint *for some Hue devices* but not for others isn't the best approach in my opinion. The images provided by the Hue Bridge currently contain images for all Hue devices.
At the moment, I would personally prefer to regularly (but "manually") update the Hue images in the zigbee-OTA repo.

@mkaay already touched on this subject here: https://github.com/Koenkk/zigbee-herdsman-converters/issues/2168

If Z2M would support multiple OTA providers (prioritizing newer images), it's possible that a clean solution could be made, as the zigbee-OTA repo could still have links to all Hue images provided by the Hue Bridge, but Z2M could still check for all devices if a (newer) image is available through the mobile endpoint. (Again, note that images provided through the mobile endpoint may differ/be a bit more "experimental" for the smaller user base.)

TL;DR:
- this approach works for some Hue lights (not all but more than in this PR)
- the images from this endpoint are for the Hue Bluetooth app and are -- in rare cases -- newer than the ones provided by the Hue Bridge (potentially more "experimental" for the small user base)
- it's nice for ZHA, deCONZ users to have a list of all Hue images in zigbee-OTA
- multiple OTA providers are not possible in Z2M => this makes it hard to create "a clean solution" for all Hue devices

**Comment by henriknil** [comment]
@TheJulianJES , thanks for the reply. Maybe support for multiple OTA providers is a feasible way forward.
I will think a bit about it.
Is there any more information available on the workings of the bridge endpoint? I assume with a rooted device you could extract the bridge-specific data for the other endpoint?

**Comment by mkaay** [comment]
The Auth for the API is individual for each bridge and I managed to root mine and reverse engineer the encryption and communication.

So we have these options in my opinion:
- Build and host an API proxy
- Automatic pull request on update via cronjob
- update check via API, but manual pull request (like right now)

From a reliability/maintenance standpoint the automatic pull request might the best. Since I also don't know how much traffic this will cause.

**Comment by henriknil** [comment]
@mkaay , sounds like a good idea.
The reason for me to try to implement was that I think there is missing an older firmware for https://www.zigbee2mqtt.io/devices/324131092621.html#philips-324131092621. Is it possible for you to add this to zigbeeOTA?

**Comment by mkaay** [comment]
If you have the imagetype the dimmer is requesting, me or @TheJulianJES can get it for you

**Comment by henriknil** [comment]
I'm not at home but from checking index.json in zigbeeOTA I pretty sure it's imageType=265

**Comment by TheJulianJES** [comment]
Is there any reason you think there's an older firmware OTA image for that dimmer? (Doesn't look like there's an older version than the one currently in the zigbee-OTA repo).

**Comment by henriknil** [comment]
> Is there any reason you think there's an older firmware OTA image for that dimmer? (Doesn't look like there's an older version than the one currently in the zigbee-OTA repo).
> 
> Previously discussed: [Koenkk/zigbee2mqtt#10139 (comment)](https://github.com/Koenkk/zigbee2mqtt/issues/10139#issuecomment-1030628109) (ZGB_100B_0109/image type 265)

It was only an assumption due to that firmware upgrades doesn't seem to work and for two other imagetypes adding an older version was the solution. But since you now checked the available version for me I will do some further investigation and possible get back with comments in [Koenkk/zigbee2mqtt#10139].
I guess some sniffing will be needed :)

**Comment by mkaay** [comment]
I already have such a script for the bridge API :)
